src/promptflow-evals/promptflow/evals/synthetic/README.md

```diff
@@ -1,66 +1,280 @@
+This document provides instructions on how to use the adversarial simulator.
 
 ## Prerequisites
 
 Ensure you have Python installed on your system. If not, you can download it from [here](https://www.python.org/downloads/).
 
 ## Installation
 
-Install the `promptflow-evals` extra while installing promptflow.
+Install the `promptflow-evals` package.
+
+`pip install promptflow-evals`
 
 ## Usage
-Set the values for `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_ENDPOINT` in the env or update this script:
 
-```
-import os
-from promptflow.evals.synthetic.qa import QADataGenerator, QAType, OutputStructure
+```python
+from promptflow.evals.synthetic import AdversarialSimulator
+import config
+from azure.identity import DefaultAzureCredential
+from typing import Any, Dict, List, Optional
+import asyncio
+
+
 
-os.environ["AZURE_OPENAI_API_KEY"] = ""
-os.environ['AZURE_OPENAI_ENDPOINT'] = ""
+project_scope = {
+    "subscription_id": config.sub,
+    "resource_group_name": config.rg,
+    "workspace_name": config.project_name,
+    "credential": DefaultAzureCredential(),
+}
 
-model_name = "gpt-4"
+async def callback(
+    messages: List[Dict],
+    stream: bool = False,
+    session_state: Any = None,
+    context: Dict[str, Any] = None
+) -> dict:
+    question = messages["messages"][0]["content"]
+    context = None
+    if 'file_content' in messages["template_parameters"]:
+        question += messages["template_parameters"]['file_content']
+    from openai import AsyncAzureOpenAI
 
-model_config = dict(
-    deployment=model_name,
-    model=model_name,
-    max_tokens=2000,
+    oai_client = AsyncAzureOpenAI(
+        api_key=config.api_key,
+        azure_endpoint=config.endpoint,
+        api_version="2023-12-01-preview",
+    )
+    try:
+        response_from_acs = await oai_client.chat.completions.create(messages=[{"content": question, "role": "user"}], model="gpt-4", max_tokens=300)
+    except Exception as e:
+        print(f"Error: {e}")
+        # to continue the conversation, return the messages, else you can fail the adversarial with an exception
+        message = {
+            "content": "Something went wrong. Check the exception e for more details.",
+            "role": "assistant",
+            "context": None,
+        }
+        messages["messages"].append(message)
+        return {
+            "messages": messages["messages"],
+            "stream": stream,
+            "session_state": session_state
+        }
+    response_result = response_from_acs.choices[0].message.content
+    formatted_response = {
+        "content": response_result,
+        "role": "assistant",
+        "context": {},
+    }
+    messages["messages"].append(formatted_response)
+    return {
+        "messages": messages["messages"],
+        "stream": stream,
+        "session_state": session_state,
+        "context": context
+    }
+
+```
+
+### Adversarial QA:
+```python
+template = "adv_qa"

```

Comment:
echo Lisa's suggestion about put template into simluator's callable parameter instead of init parameter. Also, can the template(scenario) be a enum to help user select them easier?

Reply:
Doing that will limit us in this way:
when new scenarios are added to the service, we would have to update the SDK to support them.

Reply:
so do you mean there's no code logic in SDK to handle the different template , but it only pass the template name to server?

Reply:
Yes, there’s no logic to verify/validate the value being passed to the scenario. 
if the service doesn’t respond with the right values, we show an error to the customer saying their input could be wrong.